third_party/terraform/resources/resource_google_project.go

```diff
@@ -263,7 +263,11 @@ func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error
 		}
 
 		if err = forceDeleteComputeNetwork(project.ProjectId, "default", config); err != nil {
-			return fmt.Errorf("Error deleting default network in project %s: %s", project.ProjectId, err)
+			if isGoogleApiErrorWithCode(err, 404) {
```

Comment:
Can't we instead query for the existence of a default network and skip the call to delete if it already doesn't exist?

Reply:
We could! I chose this way for three reasons:

1. Either way we'd end up with a network call, except in the most common case we'd end up with two if we checked for existence first
2. I'm not sure how the current organizational policy works (whether a network is created, then deleted, or if it's just not created at all) or if any future improvements may lead to a network being created then deleting, and I worried about race conditions
3. Returning a 404 when trying to delete something never seems like an appropriate response, as far as I can think?

Because of that, I couldn't see any benefit to it. Is there a reason you think it'd be better?

Reply:
I was just thinking about the error on the customer logs related to the delete of the inexistent network, but I'm not strongly opinionated either way.